### ISSUE TITLE
checkJs doesn't detect types of object keys passed to a function

### ISSUE THREAD

**Issue by francescoinfante** [Microsoft/TypeScript#15700]
**TypeScript Version:**  2.3.2

**Code**

```json
{
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "allowJs": true,
    "checkJs": true,
    "noEmit": true
  },
  "include": [
    "./test.js"
  ]
}

```

```js
/**
 * @param {string} a
 */
function example1(a) {
  a === 2; // error TS2365: Operator '===' cannot be applied to types 'string' and 'number'.
}

/**
 * @param {any} a
 * @param {string} a.b
 */
function example3(a) {
  a.b === 2; // no errors
}

/**
 * @param {{b: string}} a
 * @param {string} a.b
 */
function example5(a) {
  a.b === 2; // error TS2365: Operator '===' cannot be applied to types 'string' and 'number'.
}

/**
 * @param {{b: string}} a
 * @param {number} a.b
 */
function example6(a) {
  a.b === 2; // error TS2365: Operator '===' cannot be applied to types 'string' and 'number'.
}

/**
 * @param {number} a.b
 */
function example7(a) {
  a.b === 2; // error TS2339: Property 'b' does not exist on type 'number'.
}

```

**Expected behavior:**

Example 1 works correctly
I expect in example 3 that `a.b` is detected as a `string` and that error `TS2365` is shown
Example 5 works correctly
I expect in example 6 to say that it's ambiguous.
I expect in example 7 that `a` is detected as an object and not as a `number`.

**Actual behavior:**
Errors shown in the code

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/11597
